go/enclave/gas/oracle.go

```diff
@@ -44,45 +67,92 @@ func (o *oracle) EstimateL1StorageGasCost(tx *types.Transaction, block *types.He
 	return o.calculateL1Cost(block, header, encodedTx)
 }
 
-func (o *oracle) EstimateL1CostForMsg(args *gethapi.TransactionArgs, block *types.Header, header *common.BatchHeader) (*big.Int, error) {
+func (o *oracle) EstimateL1CostForMsg(args *gethapi.TransactionArgs, header *common.BatchHeader) (*big.Int, error) {
 	encoded, err := rlp.EncodeToBytes(args)
 	if err != nil {
 		return nil, err
 	}
 
-	return o.calculateL1Cost(block, header, encoded)
+	return o.calculateL1Cost(nil, header, encoded)
+}
+
+func (o *oracle) SubmitL1Block(ctx context.Context, headBlock *types.Header) error {
+	o.headMutex.Lock()
```

Comment:
why do we need the mutex, aren't blocks already mutexed?

Reply:
yeah, the writes, but not the reads.